Src/DependencyCollector/Shared/Implementation/ApplicationInsightsUrlFilter.cs

```diff
@@ -9,52 +9,33 @@ internal sealed class ApplicationInsightsUrlFilter
         internal const string TelemetryServiceEndpoint = "https://dc.services.visualstudio.com/v2/track";
         internal const string QuickPulseServiceEndpoint = "https://rt.services.visualstudio.com/QuickPulseService.svc";
 
+        internal readonly Uri TelemetryServiceEndpointUri = new Uri(TelemetryServiceEndpoint);
         private readonly TelemetryConfiguration telemetryConfiguration;
 
         private KeyValuePair<string, string> cachedEndpointLeftPart = new KeyValuePair<string, string>();
 
         public ApplicationInsightsUrlFilter(TelemetryConfiguration telemetryConfiguration)
         {
             this.telemetryConfiguration = telemetryConfiguration;
+
+            // cache EndpointLeftPart ahaed, before first dependency call
+            this.GetEndpointLeftPart();
```

Comment:
Apparently caching endpoint is an expensive operation that takes ~200 microseconds in release build on my machine.